### PR TITLE
feat:support env and appenv instruction

### PR DESCRIPTION
**Pull request by kakaZhou719**
### Describe what this PR does / why we need it
Kubefile:

```shell
FROM scratch
ENV globalKey=globalValue
APP app1 local://app1
APP app2 local://app2
APP app3 local://app3
APPENV app1 key1=value1 key2=value2
LAUNCH ["app1","app2"]
```

```json
{
  "buildClient": {
    "sealerVersion": "unknown",
    "buildahVersion": "1.27.1"
  },
  "schemaVersion": "v1alpha1",
  "type": "app-installer",
  "applications": [
    {
      "name": "app1",
      "type": "shell",
      "envs": [
        "key1=value1",
        "key2=value2",
      ],
      "files": [
        "app1.sh"
      ],
      "version": "v1"
    },
    {
      "name": "app2",
      "type": "shell",
      "files": [
        "app2.sh"
      ],
      "version": "v1"
    },
    {
      "name": "app3",
      "type": "shell",
      "files": [
        "app3.sh"
      ],
      "version": "v1"
    }
  ],
  "labels": {
    "io.buildah.version": "1.27.1",
    "io.sealer.version": "unknown"
  },
    "env": {
            "globalKey": "globalValue"
        },
  "launch": {
    "app_names": [
      "app1",
      "app2"
    ]
  }
}
```

### Does this pull request fix one issue?

Fixes https://github.com/sealerio/sealer/issues/2173

### CODE REVIEW

File: build/kubefile/parser/kubefile.go [hunk 1]
```diff
@@ -304,6 +333,75 @@ func (kp *KubefileParser) processAppCmds(node *Node, result *KubefileResult) err
...
+		}
+
+		var kv []string
+		if kv = strings.SplitN(elem, "=", 2); len(kv) != 2 {
+			continue
+		}
+		result.GlobalEnv[kv[0]] = kv[1]
+	}
+
+	return nil
```
Comment (Starnop) [inline build/kubefile/parser/kubefile.go:402]: update result.Dockerfile with global env?

File: build/kubefile/parser/kubefile.go [hunk 2]
```diff
@@ -179,6 +192,18 @@ func (kp *KubefileParser) generateResult(mainNode *Node) (*KubefileResult, error
 		}
 	}
 
+	// register app with all env list.
+	for appName, appEnv := range result.AppEnvMap {
+		app := result.Applications[appName]
```
Comment (Starnop) [inline build/kubefile/parser/kubefile.go:197]: maybe `result.Applications[appName].AppEnv = appEnv` is better

Comment (kakaZhou719) [reply build/kubefile/parser/kubefile.go:197]: this is not a copy, `result.Applications` is a interface, and i am trying to register v1app here with all env list.

**Review by Starnop — APPROVED** [review]
LGTM

**Pull request merged**